runtime/jcl/common/java_lang_Class.cpp

```diff
@@ -1837,29 +1881,51 @@ Java_java_lang_Class_getNestHostImpl(JNIEnv *env, jobject recv)
 
 	vmFuncs->internalExitVMToJNI(currentThread);
 	return result;
+#else
+	Assert_JCL_unimplemented();
+	return NULL;
+#endif /* defined(J9VM_OPT_VALHALLA_NESTMATES) */
 }
 
 jobject JNICALL
 Java_java_lang_Class_getNestMembersImpl(JNIEnv *env, jobject recv)
 {
+#if defined(J9VM_OPT_VALHALLA_NESTMATES)
 	J9VMThread *currentThread = (J9VMThread*)env;
 	J9JavaVM *vm = currentThread->javaVM;
 	J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
 	J9MemoryManagerFunctions *mmFuncs = vm->memoryManagerFunctions;
 
 	j9object_t resultObject = NULL;
 	jobject result = NULL;
+	J9ROMClass *romHostClass = NULL;
+	U_16 nestMemberCount = 0;
+	J9Class *jlClass = NULL;
+	J9Class *arrayClass = NULL;
+	J9Class *nestMember = NULL;
+
+	IDATA nestLoadFailure = J9_VISIBILITY_ALLOWED;
 
 	vmFuncs->internalEnterVMFromJNI(currentThread);
 
 	J9Class *clazz = J9VM_J9CLASS_FROM_HEAPCLASS(currentThread, J9_JNI_UNWRAP_REFERENCE(recv));
 	J9Class *nestHost = clazz->nestHost;
-	J9ROMClass *romHostClass = nestHost->romClass;
-	U_16 nestMemberCount = romHostClass->nestMemberCount;
 
-	/*  Grab the Class<?> class for result object size */
-	J9Class *jlClass = J9VMJAVALANGCLASS(vm);
-	J9Class *arrayClass = fetchArrayClass(currentThread, jlClass);
+	if (NULL == nestHost) {
+		nestLoadFailure = vmFuncs->loadAndVerifyNestHost(currentThread, clazz, FALSE);
+		if (J9_VISIBILITY_ALLOWED != nestLoadFailure) {
+			nestMember = clazz;
```

Comment:
Just wondering is this right time to exit with exception? `getNestHost` should return a nestHost or itself if there is an error.

Reply:
this is getNestMember not getNestHost. GetNestMember must throw exceptions if it fails to load every member of the nest

Reply:
Tobi said `RI` has same behaviours when running the tests in this PR as `OpenJ9`, so `GetNestMember` throws exceptions if it fails to load every member of the nest including nest host.